src/auth/strategies/google.strategy.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, Profile } from 'passport-google-oauth20';
+import { ConfigService } from '@nestjs/config';
+import { OAuthProvider } from '@prisma/client';
+
+@Injectable()
+export class GoogleStrategy extends PassportStrategy(Strategy, 'google') {
+  constructor(cfg: ConfigService) {
+    super({
+      clientID: cfg.getOrThrow<string>('GOOGLE_CLIENT_ID'),
+      clientSecret: cfg.getOrThrow<string>('GOOGLE_CLIENT_SECRET'),
+      callbackURL: cfg.getOrThrow<string>('GOOGLE_CALLBACK_URL'),
+      scope: ['profile', 'email'],
+    });
+  }
+
+  validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: Profile,
+    done: (err: any, user?: any) => void,
+  ) {
+    const emailObj = profile.emails?.[0];
+    done(null, {
+      provider: OAuthProvider.GOOGLE,
+      providerId: profile.id,
+      email: emailObj?.value,
+      emailVerified: emailObj?.verified ?? false,
+    });
+  }
```

Comment:
```suggestion
  validate(
    accessToken: string,
    refreshToken: string,
    profile: Profile,
  ) {
    const emailObj = profile.emails?.[0];
    return {
      provider: OAuthProvider.GOOGLE,
      providerId: profile.id,
      email: emailObj?.value,
      emailVerified: emailObj?.verified ?? false,
    };
  }
```
현재 GoogleStrategy에서 validate 함수 안에서 done 콜백을 직접 호출하고 있습니다.
NestJS의 PassportStrategy는 validate가 Promise/return 기반으로 동작하도록 설계되어 있고, 반환값을 알아서 done에 전달해 줍니다.

done을 직접 호출하는 것도 동작은 하지만 Nest의 의도된 흐름(리턴 기반 인증 플로우)을 깨뜨릴 수 있어서 return 방식으로 작성하는 것이 더 안전하고 NestJS 권장 패턴에 맞을 것 같습니다.